src/remote/action.go

```diff
@@ -152,14 +152,10 @@ func (c *Client) buildTestCommand(target *core.BuildTarget) (*pb.Command, error)
 // getCommand returns the appropriate command to use for a target.
 func (c *Client) getCommand(target *core.BuildTarget) string {
 	if target.IsRemoteFile {
-		// TODO(peterebden): we should handle this using the Remote Fetch API once that's available.
-		urls := make([]string, len(target.Sources))
-		for i, s := range target.Sources {
-			urls[i] = "curl -fsSLo $OUT " + s.String()
-		}
-		cmd := strings.Join(urls, " || ")
+		// This isn't a real command, but it suits us to construct a pseudo-version of one.
+		cmd := "fetch " + strings.Join(target.AllURLs(c.state.Config), " ") + " & verify " + strings.Join(target.Hashes, " ")
 		if target.IsBinary {
-			return "(" + cmd + ") && chmod +x $OUT"
+			return cmd + " binary"
 		}
```

Comment:
How does this work?

Reply:
It doesn't really. It's not a real command but it's a lot easier to generate a BuildAction and bung it in the CAS so the rest of the code can be agnostic to how the target was executed. Basically I'm just making something up that contains the important bits of the rule (the URLs and hashes etc) so it hashes uniquely.